### PR TITLE
Using abort_callback from ggml to interrupt llama computation

### PR DESCRIPTION
**Pull request by Xarbirus**
I tried to use the abort_callback mechanism from ggml to interrupt llama.cpp computations.
What do you think about this way?

### CODE REVIEW

**Comment by Xarbirus** [comment]
Waiting for [this](https://github.com/ggerganov/ggml/pull/725) PR in ggml library.

**Comment by ggerganov** [comment]
This can be rebased now

**Comment by Xarbirus** [comment]
> This can be rebased now

Thank you, I will finish this PR next week.

**Comment by Xarbirus** [comment]
@ggerganov
rebased, ready to be merged.

File: llama.h [hunk 1]
```diff
@@ -256,6 +256,9 @@ extern "C" {
         bool embedding;   // embedding mode only
         bool offload_kqv; // whether to offload the KQV ops (including the KV cache) to GPU
         bool do_pooling;  // whether to pool (sum) embedding results by sequence id (ignored if no pooling layer)
+
+        ggml_abort_callback abort_callback;
+        void *              abort_callback_data;
```
Comment (slaren) [inline llama.h:261]: A brief comment explaining what is the use of the abort callback would be nice.

**Review by slaren — APPROVED** [review]

**Review by ggerganov — APPROVED** [review]

**Pull request merged**